install_script.sh.template

```diff
@@ -378,6 +378,212 @@ function ensure_config_file_exists() {
     fi
 }
 
+function report_installer_telemetry() {
+    local trace_id="$1"
+    local os="$2"
+    local distribution="$3"
+    local start_time="$4"
+    local exit_code="$5"
+    local install_stdout="$6"
+    local install_stderr="$7"
+
+    local time_now_seconds
+    local time_now
+    local time_since_start
+    local log_lines_stderr
+    local log_lines_stdout
+    local json_logs
+    local telemetry_trace
+    local telemetry_logs
+
+    echo -e "\033[34m\n* Reporting installer telemetry\n\033[0m"
+
+    if [ "$DD_INSTRUMENTATION_TELEMETRY_ENABLED" == "false" ] || \
+      [ "$site" == "ddog-gov.com" ] || \
+      [ -z "${apikey}" ] || \
+      [ -z "$telemetry_url" ]; then
+      return
+    fi
+
+    time_now_seconds=$(date +%s)
```

Comment:
if any of those command fail, wont the above `set -e` trigger an end of script?
in which case a || defaultValue would protect or maybe wrapping the install command with set +e, set -e

Reply:
Isolated the telemetry in a subshell to nullify the risk